core/base/implicitTriangulation/ImplicitTriangulation.h

```diff
@@ -918,10 +945,165 @@ namespace ttk {
     bool TTK_TRIANGULATION_INTERNAL(isEdgeOnBoundary)(
       const SimplexId &edgeId) const override;
 
-    int TTK_TRIANGULATION_INTERNAL(getVertexNeighbor)(
+    inline int TTK_TRIANGULATION_INTERNAL(getVertexNeighbor)(
       const SimplexId &vertexId,
       const int &localNeighborId,
-      SimplexId &neighborId) const override;
+      SimplexId &neighborId) const final {
+
+      if(dimensionality_ == 3) {
+        const ttk::SimplexId z = vertexId / vshift_[1];
+        const auto idx2 = vertexId - (z * vshift_[1]);
+        const ttk::SimplexId y = idx2 / vshift_[0];
+        const ttk::SimplexId x = idx2 % vshift_[0];
+
+        int key = (x == 0              ? 1
+                   : x == nbvoxels_[0] ? 2
+                                       : 0)
+                  + (y == 0              ? 4
+                     : y == nbvoxels_[1] ? 8
+                                         : 0)
+                  + (z == 0              ? 16
+                     : z == nbvoxels_[2] ? 32
+                                         : 0);
+
+        neighborId = vertexId + neighborIndexLUT[key * 14 + localNeighborId];
+
+        return 0;
+      } else {
+#ifndef TTK_ENABLE_KAMIKAZE
```

Comment:
same comment as above (move the `ifndef` block at the beginning)

Reply:
fixed